andela/valinor-ah-backend
# #161291014 Implement receive notifcation feature

#### What does this PR do?

Implement receive notification feature

#### Description of Task to be completed?

- Write tests for NotifcationEvent model
- Write tests for NotificationController
- Add Notification model
- Add NotificationEvent model
- Implement notification via email
- Add NotificationController
- Add opt-in and opt-out functionality
- Get all notification for a user functionality

#### How should this be manually tested?
- clone the repository, cd into it, checkout this branch feature/161291014/receive-notification
- set up your .env variables and local database
- run npm install to install required packages
- run npm start to start up the server
- use postman(or any api testing tool) to test the following routes route
- create a user using the route POST `http://localhost:3000/api/v1/users/signup` providing necessary request body.
- modify account status PATCH `http://localhost:3000/api/v1/users/:userId/account/:action` providing the token in the Authorization header as shown in screen shots below.
- the action param can either be activate, deactivate or delete.
- when you set action to delete, verify that you get an email in the address provided at signup
- click this link and verify that account is deleted by checking the db
- hit `http://localhost:3000/api/v1/users/follow/2`

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#161291014](https://www.pivotaltracker.com/story/show/161291014)

#### Screenshots (if appropriate)
<img width="826" alt="screen shot 2018-11-19 at 4 20 22 pm" src="https://user-images.githubusercontent.com/25647480/48716202-183f3a80-ec17-11e8-9a81-0fd71c921e69.png">

- New Comment Notification Screenshot
<img width="712" alt="screen shot 2018-11-23 at 5 58 23 pm" src="https://user-images.githubusercontent.com/25647480/48954480-e8ab6d80-ef49-11e8-9fe3-0ac6cf9010ff.png">

- New Follow Notification Screenshot
<img width="711" alt="screen shot 2018-11-23 at 5 58 02 pm" src="https://user-images.githubusercontent.com/25647480/48954482-ee08b800-ef49-11e8-8181-3f7f3c559493.png">

#### Questions:

N/A
